marlowe-playground-client/src/Wallet.purs

```diff
@@ -0,0 +1,1301 @@
+            <> tableRow
+                { title: "Payments"
+                , emptyMessage: "No payments have been recorded"
+                , columns: ("Party" /\ "Asset" /\ mempty)
+                , rowData: paymentsData
+                }
+            <> tableRow
+                { title: "Let Bindings"
+                , emptyMessage: "No values have been bound"
+                , columns: ("Identifier" /\ "Value" /\ mempty)
+                , rowData: bindingsData
+                }
+        )
+    ]
+  where
+  contractMaxTime Nothing = "Closed"
+
+  contractMaxTime (Just contract) = let t = maxTime contract in if t == zero then "Closed" else show t
+
+  warnings = state ^. (_currentLoadedMarloweState <<< _transactionWarnings)
+
+  warningsRow =
+    if Array.null warnings then
+      []
+    else
+      (headerRow "Warnings" ("type" /\ "details" /\ mempty)) <> foldMap displayWarning' warnings
+
+  errorRow = case state ^? (_currentLoadedMarloweState <<< _transactionError) of
+    Nothing -> []
+    Just error ->
+      if isNothing error then
+        []
+      else
+        (headerRow "Errors" ("details" /\ mempty /\ mempty)) <> displayError error
+
+  displayError Nothing = []
+
+  displayError (Just err) =
+    [ div [ classes [ rTableCell, first ] ] []
+    , div [ classes [ rTableCell, ClassName "Rtable-single-column-row" ] ] [ text $ show err ]
+    ]
+
+  accountsData =
+    let
+      (accounts :: Array _) = state ^. (_currentLoadedMarloweState <<< _state <<< _accounts <<< to Map.toUnfoldable)
+
+      asTuple (Tuple (Tuple (AccountId accountNumber accountOwner) token) value) = show accountNumber /\ stripParens (show accountOwner) /\ (show value <> " " <> shortTokenString token)
+    in
+      map asTuple accounts
+
+  shortTokenString (Token "" "") = "Ada"
+
+  shortTokenString (Token currSym tokName) = currSym <> " " <> tokName
+
+  choicesData =
+    let
+      (choices :: Array _) = state ^. (_currentLoadedMarloweState <<< _state <<< _choices <<< to Map.toUnfoldable)
+
+      asTuple (Tuple (ChoiceId choiceName choiceOwner) value) = show choiceName /\ stripParens (show choiceOwner) /\ show value
+    in
+      map asTuple choices
+
+  paymentsData =
+    let
+      payments = state ^. (_currentLoadedMarloweState <<< _payments)
+
+      asTuple :: Payment -> Array (String /\ String /\ String)
+      asTuple (Payment party (Assets mon)) =
+        concatMap
+          ( \(Tuple currencySymbol tokenMap) ->
+              ( map
+                  ( \(Tuple tokenName value) ->
+                      stripParens (show party) /\ (show value <> " " <> shortTokenString (Token currencySymbol tokenName)) /\ mempty
+                  )
+                  (Map.toUnfoldable tokenMap)
+              )
+          )
+          (Map.toUnfoldable mon)
+    in
+      concatMap asTuple payments
+
+  bindingsData =
+    let
+      (bindings :: Array _) = state ^. (_currentLoadedMarloweState <<< _state <<< _boundValues <<< to Map.toUnfoldable)
```

Comment:
`toArrayOf ...`?

Reply:
doesn't seem to give me the Tuple I need